EiDiA_backend/src/controllers/search.js

```diff
@@ -1,14 +1,84 @@
 "use strict";
 
+const FileTypes = require('../constants').fileTypes;
+const FileActions = require('../constants').actions;
+const RecordModel = require('../models/record');
 
 const basicSearch = (req, res) => {
     res.status(200).json({response: "dummy response"});
 };
 
 const advancedSearch = (req, res) => {
-    res.status(200).json({response: "dummy response"});
+    const fileArray = [];
+    getSearchTable(fileArray, req.userId)
+        .then(table => {
+            res.status(200).json({response: table});
+        })
+        .catch(error => {
+            res.status(400).json({
+                error: error,
+                message: error.message,
+            });
+        });
 };
 
+function getSearchTable(fileArray, userId) {
+    let recordIds = [];
+    fileArray.map(file => {
+        if (!recordIds.includes(file.recordId)) {
+            recordIds.push(file.recordId);
+        }
+    });
+
+    return new Promise((resolve, reject) => {
+        Promise.all(recordIds.map(recordId => getRecordFolderElements(recordId)))
+            .then(folders => {
+                fileArray.map(document => {
+                    let fileActions = [FileActions.EDIT, FileActions.DOWNLOAD];
+                    if (document.createdBy === userId) {
+                        fileActions.push(FileActions.DELETE);
+                    }
+                    folders.push({
+                        parentId: document.recordId,
+                        id: document._id,
+                        active: false,
+                        type: document.fileType,
+                        name: document.name,
+                        dateCreation: document.createdOnDate,
+                        dateModification: document.lastModifiedOnDate,
+                        comment: document.comment,
+                        actions: fileActions,
+                    });
+                });
+                return resolve(folders);
+            })
+            .catch(error => reject(error));
+    });
+}
+
+function getRecordFolderElements(recordId) {
+    return new Promise((resolve, reject) => {
+        RecordModel.findById(recordId, {}, {}, (err, record) => {
+            if (err) {
+                console.log(err);
+                reject(err);
+            }
+            let folderElement = record === undefined ? null : {
+                parentId: 0,
+                id: recordId,
+                active: false,
```

Comment:
```suggestion
                active: true,
```
ich will ja dass meine Ordner by default geöffnet dargestellt werden